scripts/plot_radiosonde_data.py

```diff
@@ -112,6 +113,30 @@ def data4skewt(direction, data):
         u, v = mpcalc.wind_components(wind_speed, wind_dir)
     return p, T, Td, wind_speed, wind_dir, u, v
 
+def calculate_max_distance(ds):
+    latitudes = ds['lat'].values
+    longitudes = ds['lon'].values
+    
+    # Ensure there are no NaN values by filtering them out
+    valid_indices = ~np.isnan(latitudes) & ~np.isnan(longitudes)
+    latitudes = latitudes[valid_indices]
+    longitudes = longitudes[valid_indices]
+    
+    # Skip soundings with insufficient valid data
+    if len(latitudes) < 2 or len(longitudes) < 2:
+        print(f"Not enough valid data for sounding {sounding}. Skipping.")
+        return None
+    
+    # Get the surface (initial) latitude and longitude
+    surface_lat = latitudes[0]
+    surface_lon = longitudes[0]
+    
+    # Calculate the distance from the surface point to each other point
+    distances = np.array([geodesic((surface_lat, surface_lon), (lat, lon)).kilometers 
+                          for lat, lon in zip(latitudes, longitudes)])
+    
```

Comment:
Could you please run the pre-commit hooks. The formatting here does not comply with the set standard.

You can do:
```
pip install pre-commit
pre-commit install
pre-commit run --all
git add <the changed files>
git commit
```